image/templates/helm/shared/templates/_scanner-v4_init.tpl.htpl

```diff
@@ -0,0 +1,43 @@
+{{/*
+  srox.scannerV4Init . $scannerV4Config
+
+  Initializes the Scanner v4 configuration. The scanner chart has two modes: Indexer and Matcher.
+  In Indexer mode, the Scanner pulls images and analyzes them to determine the base OS and installed packages
+  (i.e. it indexes the images).
+  In Matcher mode, the Scanner matches the found packages to known vulnerabilities to produce a Vulnerability Report.
+  Both modes require access to a PostgreSQL database.
+
+  StackRox's Central service has two Scanner deployments: a Scanner running in Indexer mode and another
+  running in Matcher mode. In this context, the Helm chart can create its own certificates.
+
+  StackRox's Secured Cluster services may deploy the Scanner in Indexer mode, only.
+  This would be done to access registries inaccessible to the Central cluster.
+  In this context, the Helm chart does not generate its own certificates.
+
+  $scannerV4Config contains all values which are configured by the user. The structures can be viewed in the respective
+  config-shape. See internal/scanner-v4-config-shape.yaml.
+   */}}
+
+{{ define "srox.scannerV4Init" }}
+
+{{ $ := index . 0 }}
+{{ $scannerV4Cfg := index . 1 }}
+
+{{ if not $scannerV4Cfg.indexer.disable }}
+    {{ include "srox.configureImage" (list $ $scannerV4Cfg.indexer.image) }}
+    {{ include "srox.configureImage" (list $ $scannerV4Cfg.db.image) }}
+{{ end }}
+{{ if not $scannerV4Cfg.matcher.disable }}
```

Comment:
There may be a better way to do this, and this is definitely not complete. The purpose of this is to do three things:

1. Configure the images
2. Configure the TLS certs
3. Configure the db password

For the TLS certs, take a look at the comments at the top of the file. We are *not* to set them in the SecuredCluster. So, that is why I set the INDEXER cert when matcher is enabled. That's because we expect the matcher to only be deployed in CentralCluster and we expect both Indexer and Matcher to exist. Feel free to change the logic for this (and add the MATCHER cert...).